lib/_http_outgoing.js

```diff
@@ -635,30 +637,58 @@ ObjectDefineProperty(OutgoingMessage.prototype, 'writableEnded', {
 
 const crlf_buf = Buffer.from('\r\n');
 OutgoingMessage.prototype.write = function write(chunk, encoding, callback) {
+  if (typeof encoding === 'function') {
+    callback = encoding;
+    encoding = null;
+  }
+
   const ret = write_(this, chunk, encoding, callback, false);
   if (!ret)
     this[kNeedDrain] = true;
   return ret;
 };
 
-function writeAfterEnd(msg, callback) {
-  const err = new ERR_STREAM_WRITE_AFTER_END();
+function errorOrDestroy(msg, err) {
   const triggerAsyncId = msg.socket ? msg.socket[async_id_symbol] : undefined;
   defaultTriggerAsyncIdScope(triggerAsyncId,
                              process.nextTick,
-                             writeAfterEndNT,
+                             emitErrorNt,
                              msg,
-                             err,
-                             callback);
+                             err);
+}
```

Comment:
@mcollina @addaleax This `defaultTriggerAsyncIdScope` logic. Should we also have that in `/internal/stream/destroy` if the stream has `async_id_symbol`? I'm unsure what this actually does but got curious.

Reply:
No, you should not move that. This falls into the realm of async_hooks and it's related to the fact that multiple request/response flows can pass through the same underlining socket.